sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/NetAppVolumeProvider.java

```diff
@@ -0,0 +1,69 @@
+package com.mesosphere.sdk.offer.evaluate;
+
+import com.mesosphere.sdk.scheduler.SchedulerUtils;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * This class contains NetApp-specific logic for handling volume name and volume's options.
+ */
+public class NetAppVolumeProvider implements ExternalVolumeProvider {
+
+  private static final String SHARED_KEY = "shared";
+
+  String volumeName;
+
+  Map<String, String> driverOptions;
+
+  public NetAppVolumeProvider(
+      String serviceName,
+      Optional<String> providedVolumeName,
+      String podType,
+      int podIndex,
+      Map<String, String> driverOptions)
+  {
+
+    String volumeNameUnescaped;
+    if (providedVolumeName.isPresent()) {
+      volumeNameUnescaped = providedVolumeName.get();
+    } else {
+      volumeNameUnescaped = providedVolumeName.filter(name -> !name.isEmpty()).orElse(serviceName) + '_' + podType;
+    }
+
+    String volumeNameEscaped = SchedulerUtils.withEscapedSlashes(volumeNameUnescaped);
+
+    if (driverOptions.containsKey(SHARED_KEY) && driverOptions.get(SHARED_KEY).equals("true")) {
+      // If it is a shared volume, reset the volume name to
+      // not have the pod index, so that all tasks get the
+      // same volume
+      this.volumeName = volumeNameEscaped;
+    } else {
+      this.volumeName = volumeNameEscaped + "_" + podIndex;
+    }
+
+    Map<String, String> options = new HashMap<>(driverOptions);
+    options.put("name", this.volumeName);
+    // Favor creating volumes on the local node
+    options.put("nodes", "LocalNode");
+
+    this.volumeName = options.keySet().stream()
+        .map(key -> key + "=" + options.get(key))
+        .collect(Collectors.joining(";"));
```

Comment:
@farhan5900 these are the options for Portworx Storage Driver, NetApp has different options documented here:
https://netappdvp.readthedocs.io/en/stable/use/index.html#volume-driver-cli-options

Especially ones like `SHARED_KEY` are invalid I'd assume.